src/gluonts/model/rotbaum/_preprocess.py

```diff
@@ -352,13 +426,18 @@ def make_features(self, time_series: Dict, starting_index: int) -> List:
             else []
         )
 
+        # these two assertions check that the categorical features are encoded
+        np_feat_static_cat = np.array(feat_static_cat)
         assert (not feat_static_cat) or all(
-            [(np.floor(elem) == elem) for elem in feat_static_cat]
-        )  # asserts that the categorical features are encoded
+            np.floor(np_feat_static_cat) == np_feat_static_cat
+        )
+        del np_feat_static_cat
 
-        assert (not feat_static_cat) or all(
-            [(np.floor(elem) == elem) for elem in feat_dynamic_cat]
-        )  # asserts that the categorical features are encoded
+        np_feat_dynamic_cat = np.array(feat_dynamic_cat)
+        assert (not feat_dynamic_cat) or all(
+            np.floor(np_feat_dynamic_cat) == np_feat_dynamic_cat
+        )
+        del np_feat_dynamic_cat
```

Comment:
What's the purpose of this line?

Reply:
This just ensures that all the dynamic categorical variables are label encoded and are not strings or any other form that a categorical variable can take